### DISCUSSION TITLE
Vert.x registry.config file genertaion

### DISCUSSION THREAD

**Discussion by shachindrasingh** [apiman/apiman#2139] [Help]
Hi,
how can we generate the registry.config automatically . is any way to generate that file. or we need to create that file manually ?

**Comment by msavy** [comment]
Please check out the docs:

https://www.apiman.io/apiman-docs/installation-guide/2.2.3.Final/manager/configuration.html#_custom_api_catalog

I'm not sure what your requirement is, but it may require you to implement your own plugin.

**Reply by shachindrasingh** [reply]
Thanks , i will check the detail let you know in case of any issue.

**Reply by shachindrasingh** [reply]
My requirement is very simple . i want to use the vert.x conf-headless-es.json config file . for this we need the provide the Registry file detail("io.apiman.gateway.engine.vertx.polling.URILoadingRegistry").  So how can we generate the config.json file for registry.
https://www.apiman.io/apiman-docs/installation-guide/2.2.3.Final/registries-and-components/headless.html

**Reply by msavy** [reply]
If you want to use the headless registry, then you need to generate the JSON yourself using your own tooling/build process. It's meant for more advanced use-cases where the Apiman Manager isn't required and you want to do everything via configuration.

Another option worth considering is the Apiman CLI -- https://github.com/apiman/apiman-cli
